plugin-server/src/worker/vm/upgrades/historical-export/export-historical-events-v2.ts

```diff
@@ -569,6 +561,66 @@ export function addHistoricalEventsExportCapabilityV2(
         }
     }
 
+    async function handleFetchError(
+        error: Error,
+        params: ExportParams,
+        payload: ExportHistoricalEventsJobPayload
+    ): Promise<void> {
+        if (error instanceof RetryError && payload.retriesPerformedSoFar + 1 < hub.HISTORICAL_EXPORTS_MAX_RETRY_COUNT) {
+            const nextRetrySeconds = retryDelaySeconds(payload.retriesPerformedSoFar)
```

Comment:
If this goes beyond 10 minutes it could lead to multiple instances of the same job running.

Reply:
So, we should add an upper limit on `hub.HISTORICAL_EXPORTS_MAX_RETRY_COUNT`.

Reply:
Indeed wee the note in the description. I think we already have this issue with the existing retry logic, which is much more likely to be triggering. I'd rather solve that separately if that's ok?